src/binary_tree_partition.jl

```diff
@@ -219,7 +227,12 @@ function partition(
     end
   end
   tn_deltas = ITensorNetwork(vcat(output_deltas_vector...))
-  return partition(ITensorNetwork{Any}(disjoint_union(out_tn, tn_deltas)), subgraph_vs)
+  par = partition(ITensorNetwork{Any}(disjoint_union(out_tn, tn_deltas)), subgraph_vs)
```

Comment:
Was there an issue using `ITensorNetwork` instead of `ITensorNetwork{Any}`, i.e. let it try to infer the vertex type?

Reply:
The reason I force it to be `ITensorNetwork{Any}` is that in `approx_itensornetwork` I update each partition inplace during the density matrix algorithm, and that update sometimes will assign a new itensornetwork type to each partition.

But it's right that forcing the type here doesn't make sense. I made a modification to update the datatype of partition inside `approx_itensornetwork` right before the update of density matrix algorithm, so the problem should be fixed now.